subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -553,23 +576,53 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+int ztest_list_tests(void)
```

Comment:
Can you document the need for this? I would prefer to have this as a separate PR.

Reply:
I feel this pairs well with providing test cases via command line.  The output is in proper format for copy/pasting on command line.